src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1017,6 +1021,16 @@ public void RemoveOutputPath(string outputPath)
             _workspace.RemoveProjectOutputPath(Id, outputPath);
         }
 
+        public void ReorderSourceFiles(ImmutableArray<string> filePaths)
+        {
+            if (filePaths.IsEmpty)
```

Comment:
Shouldn't we validate the correctness of the array here now? The assumption here is stuff is validated up front rather than corrupting the application of the batch later.

Reply:
We could add some more validation. Most of the validation is in UpdateDocumentsOrder; there it is checking the document ids. Perhaps we need similar validation here.